test/cereal_test.exs

```diff
@@ -2,7 +2,50 @@ defmodule CerealTest do
   use ExUnit.Case
   doctest Cereal
 
-  test "the truth" do
-    assert 1 + 1 == 2
+  defmodule UserSerializer do
+    use Cereal.Serializer
+    attributes [:name, :property_that_needs_context]
+
+    def property_that_needs_context(%{_context: context}, _) do
+      "My context is comment with id #{context.id}"
+    end
+  end
+
+  defmodule CommentSerializer do
+    use Cereal.Serializer
+    attributes [:text, :property_that_needs_context]
+    has_one :user, serializer: UserSerializer, include_context: true, include: true
+
+    def property_that_needs_context(%{_context: context}, _) do
+      "My context is nil? #{is_nil(context)}"
+    end
+  end
+
+  defmodule ArticleSerializer do
+    use Cereal.Serializer
+    attributes [:name]
+    has_one :author, serializer: UserSerializer, include: false
+    has_many :comments, serializer: CommentSerializer, include: true
+  end
+
+  describe "#serialize/3" do
+    test "will include relations if include is true" do
+      author = %TestModel.User{id: 1, name: "Johnny Test"}
+      comments = [%TestModel.Comment{id: 1, text: "A comment"}]
+      article = %TestModel.Article{id: 1, name: "Article 1", comments: comments, author: author}
+      serialized = Cereal.serialize(ArticleSerializer, article, %{})
+      assert length(serialized.comments) == 1
+      refute Map.get(serialized, :author)
+    end
+
+    test "will only include the _context if include_context is true" do
+      user = %TestModel.User{id: 2, name: "Jane Test"}
+      comments = [%TestModel.Comment{id: 5, text: "A comment", user: user}]
+      article = %TestModel.Article{id: 1, name: "Article 1", comments: comments}
+      serialized = Cereal.serialize(ArticleSerializer, article, %{})
+      [comment] = serialized.comments
+      assert comment.property_that_needs_context == "My context is nil? true"
```

Comment:
How was this context-driven attribute successfully serialized for the comment? The `has_many :comments` definition on the parent Article didn't include the `include_context: true`. Also though, I wonder if that option is even necessary: maybe we just always include the parent context for use?